File: tests/adapter-fixtures/kotlin/calls.kt
package fixtures

import kotlin.collections.List

class Service {
    fun plainCall() {
        helper()
    }

    fun memberCall(obj: Worker) {
        obj.run()
    }

    fun chainedCall(registry: Registry) {
        registry.adapter().execute()
    }

    fun safeCall(maybe: Worker?) {
        maybe?.run()
    }

    fun packageCall() {
        kotlin.io.println("hi")
    }
}

class Worker {
    fun run() {}
}

class Registry {
    fun adapter(): Adapter = Adapter()
}

class Adapter {
    fun execute() {}
}

fun helper() {}

File: tests/adapter-fixtures/swift/calls.swift
import Foundation

class Service {
    func plainCall() {
        helper()
    }

    func memberCall(obj: Worker) {
        obj.run()
    }

    func chainedCall(registry: Registry) {
        registry.adapter().execute()
    }

    func selfCall() {
        self.helper()
    }
}

class Worker {
    func run() {}
}

class Registry {
    func adapter() -> Adapter { return Adapter() }
}

class Adapter {
    func execute() {}
}

func helper() {}
